### PR TITLE
NIGH-73 Regex Token Exclusion

### PR DESCRIPTION
**Pull request by alan20854**
Updating how we're doing token exclusion from hard string matching to regex

### CODE REVIEW

**Comment by alan20854** [comment]
more tests to be added

**Comment by alan20854** [comment]
note: need to rebase off brad's changes

File: internal/clients/nightfall/nightfall_internal_test.go [hunk 1]
```diff
@@ -227,15 +231,41 @@ func TestCreateCommentsFromScanResp(t *testing.T) {
...
 				},
 			},
-			haveTokenExclusionSet: tokenExclusionSet,
+			haveTokenExclusionList: tokenExclusionList,
 			want: []*diffreviewer.Comment{
 				createComment(creditCardResponse),
 			},
 			desc: "single credit card excluded",
 		},
+		{
```
Comment (alan20854) [inline internal/clients/nightfall/nightfall_internal_test.go:240]: added test where exclusion list is `["4242-4242-4242-[0-9]{4}", "^127\\."]`, and detectors are `CREDIT_CARD` and `IP_ADDRESS`

**Review by bradleymcallister97 — APPROVED** [review]
Nice!

I'm not exactly sure where this would go, but I think it might be worthwhile to put a comment somewhere saying that `TokenExclusionList` supports both exact matches and regex.

**Comment by alan20854** [comment]
> Nice!
> 
> I'm not exactly sure where this would go, but I think it might be worthwhile to put a comment somewhere saying that `TokenExclusionList` supports both exact matches and regex.

I'll make a note - It probably makes sense to put it in the readme.

File: internal/clients/nightfall/nightfall.go [hunk 2]
```diff
@@ -195,11 +194,16 @@ func createCommentsFromScanResp(
...
 
-func isFindingInTokenExclusionSet(fragment string, tokenExclusionSet *strset.Set) bool {
-	if tokenExclusionSet == nil {
+func isFindingInTokenExclusionList(fragment string, tokenExclusionList []string) bool {
+	if tokenExclusionList == nil {
 		return false
 	}
-	return tokenExclusionSet.Has(fragment)
+	for _, pattern := range tokenExclusionList {
+		if matched, _ := regexp.MatchString(pattern, fragment); matched {
```
Comment (bradleymcallister97) [inline internal/clients/nightfall/nightfall.go:202]: Seems like we should be able to use the [matchRegex](https://github.com/nightfallai/jenkins_test/pull/11/files#diff-d8a0a87ad0f085289ed44c2ba1206b24R361) func from your other diff

Comment (alan20854) [reply internal/clients/nightfall/nightfall.go:202]: yup i'll make sure to rebase

**Review by henkejosh — APPROVED** [review]

**Pull request merged**